src/Database/LSMTree/Internal/RawPage.hs

```diff
@@ -206,7 +240,7 @@ rawPageOverflowPages :: RawPage -> Int
 rawPageOverflowPages page
   | rawPageNumKeys page == 1
   , let (_, end) = rawPageValueOffsets1 page
-  = roundUpToPageSize (fromIntegral end)
+  = fromIntegral (ceilDivPageSize end - 1)  -- don't count the first page
```

Comment:
That one cost me quite a bit of time :sweat_smile:

Reply:
Did you add a test for this?

Reply:
It was only used in one function (https://github.com/IntersectMBO/lsm-tree/pull/161/files#diff-1e5323f67ff74b2afc496ab7855eb94e532d707b72750fd11d0c6fcbe78e9b0cR293), which was unused and untested in that PR.
I was waiting for that function to become available for run merging, so at least it was clear that it would immediately become tested indirectly. Still, would be nice to test it *directly* in the tests for run construction. I could do that and then open a separate PR with the test + fix?